File: app/src/main/java/com/abdessamad/orbyt/data/models/Task.kt
package com.abdessamad.orbyt.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

enum class TaskType {
    SINGLE,
    RECURRING
}

enum class TaskPriority {
    LOW,
    MEDIUM,
    HIGH
}

enum class TaskStatus {
    PENDING,
    IN_PROGRESS,
    COMPLETED
}

@Entity(tableName = "tasks")
data class Task(
    @PrimaryKey
    val id: UUID = UUID.randomUUID(),
    val title: String,
    val description: String?,
    val type: TaskType,
    val priority: TaskPriority,
    val status: TaskStatus,
    val dueDate: Date?,
    val nebulaId: UUID?,
    val createdAt: Date = Date(),
    val updatedAt: Date = Date()
)

File: fastapi/app/models/task.py
from sqlalchemy import Column, String, DateTime, Boolean, Integer, Float, Enum
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from app.database import Base
from app.schemas.task import TaskType, TaskPriority, TaskStatus

class Task(Base):
    __tablename__ = "tasks"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title = Column(String, nullable=False)
    description = Column(String)
    type = Column(Enum(TaskType), nullable=False)
    priority = Column(Enum(TaskPriority), nullable=False)
    status = Column(Enum(TaskStatus), nullable=False)
    due_date = Column(DateTime)
    nebula_id = Column(UUID(as_uuid=True))
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
